app/controllers/comments_controller.rb

```diff
@@ -8,10 +10,26 @@ def create
     redirect_to share_path(@share)
   end
 
+  def destroy
+   return render_not_found if @comment.blank?
+   return render_not_found(:forbidden) if @comment.user != current_user
+   @comment.destroy
+   flash[:notice] = "Your comment was deleted."
+   redirect_to :back
+  end
+
   private
 
-  def comment_params
-    params.require(:comment).permit(:body)
-  end
+    def comment_params
+      params.require(:comment).permit(:body)
+    end
+
+    def set_comment
+      @comment = Comment.find(params[:id])
```

Comment:
I generally like to avoid this pattern, setting and using instance variables aggressively in controllers like this can make the controllers a bit more fragile.  It's also worth noting, since `find` raises an exception when an item is not found, line 14 won't ever execute. 

If this changes to `Comment.find_by_id(params[:id])`, then it would set `@comment` to `nil`.